src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -0,0 +1,311 @@
+                data.TraceId.AsSentryId(),
+                data.OperationName,
+                data.DisplayName,
+                null,
+                null)
+            {
+                Instrumenter = Instrumenter.OpenTelemetry
+            };
+
+            var span = (SpanTracer)parentSpan.StartChild(context);
+            span.StartTimestamp = data.StartTimeUtc;
+            _map[data.SpanId] = span;
+        }
+        else
+        {
+            // If a parent exists at all, then copy its sampling decision.
+            bool? isSampled = data.HasRemoteParent ? data.Recorded : null;
+
+            // No parent span found - start a new transaction
+            var transactionContext = new TransactionContext(
+                data.SpanId.AsSentrySpanId(),
+                data.ParentSpanId.AsSentrySpanId(),
+                data.TraceId.AsSentryId(),
+                data.DisplayName,
+                data.OperationName,
+                data.DisplayName,
+                null,
+                isSampled,
+                isSampled)
+            {
+                Instrumenter = Instrumenter.OpenTelemetry
+            };
+
+            var baggageHeader = BaggageHeader.Create(data.Baggage.WithValues());
+            var dynamicSamplingContext = baggageHeader.CreateDynamicSamplingContext();
+            var transaction = (TransactionTracer)_hub.StartTransaction(
+                transactionContext, new Dictionary<string, object?>(), dynamicSamplingContext
+                );
+            transaction.StartTimestamp = data.StartTimeUtc;
+            _map[data.SpanId] = transaction;
+        }
+    }
+
+    /// <inheritdoc />
+    public override void OnEnd(Activity data)
+    {
+        // Make a dictionary of the attributes (aka "tags") for faster lookup when used throughout the processor.
+        var attributes = data.TagObjects.ToDictionary();
+
+        if (attributes.TryGetTypedValue("http.url", out string url) && _hub.IsSentryRequest(url))
+        {
+            // TODO: will this leave the span dangling?
+            _map.TryRemove(data.SpanId, out _);
+            return;
+        }
+
+        if (!_map.TryGetValue(data.SpanId, out var span))
+        {
+            _options?.DiagnosticLogger?.LogDebug("Span not found for SpanId: {0}", data.SpanId);
+            return;
+        }
+
+        var (operation, description, source) = ParseOtelSpanDescription(data, attributes);
+        span.Operation = operation;
+        span.Description = description;
+
+        if (span is TransactionTracer transaction)
+        {
+            transaction.Name = description;
+            transaction.NameSource = source;
+
+            // Use the end timestamp from the activity data.
+            transaction.EndTimestamp = data.StartTimeUtc + data.Duration;
+
+            // Transactions set otel attributes (and resource attributes) as context.
+            transaction.Contexts["otel"] = GetOtelContext(attributes);
```

Comment:
Do we define what `otel` context is? We could add types for it like we do `OperatingSystem`, etc.

Reply:
We kind of do. 
- This is the the [OpenTelemetry Support spec/doc](https://develop.sentry.dev/sdk/performance/opentelemetry/#step-5-add-opentelemetry-context).
- This is [our implementation](https://github.com/getsentry/sentry-dotnet/blob/d29723db09bcdf4d7f085248f092665b0499a3e7/src/Sentry.OpenTelemetry/SentrySpanProcessor.cs#L260) which, as you point out, is just a Dictionary.

At the moment the only place we use this is in the `SentrySpanProcessor`, to ensure this information gets sent to Sentry and recorded with the Transaction... I think this is the simplest way to do it for the time being. If we ever need to access some of the information in the `otel` context for some other reason, it might be worth wrapping it in some strong typing and making it available on the [Contexts](https://github.com/getsentry/sentry-dotnet/blob/d29723db09bcdf4d7f085248f092665b0499a3e7/src/Sentry/Contexts.cs#L14) like we do with `Browser`, `Devices`, `OperatingSystem` etc.

Or do you think we should expose it on that class already, to make it clear that it is available? I guess it's not obvious that information might be there to anyone browsing the code at the moment, unless they were familiar with this pull request or Sentry's OpenTelemetry Support spec/document.